miniwallet/deploy/101_upgrade_miniWallet.ts

```diff
@@ -0,0 +1,57 @@
+import config from '../config'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { DeployFunction } from 'hardhat-deploy/types'
+import { ethers } from 'hardhat'
+
+const OPERATOR_ROLE = ethers.utils.id('OPERATOR_ROLE')
+
+const deployFunction: DeployFunction = async function (
+  hre: HardhatRuntimeEnvironment
+) {
+  const { deployments, getNamedAccounts, getChainId } = hre
+  const { deploy } = deployments
+  const { deployer } = await getNamedAccounts()
+  const chainId = await getChainId()
+  console.log(`chainId: ${chainId}`)
+
+  const deployedContract = await deploy('MiniWallet', {
+    contract: 'MiniWallet_v2',
+    from: deployer,
+    proxy: {
+      owner: deployer,
+      proxyContract: 'EIP173Proxy'
```

Comment:
same issue here, not sure where eip173proxy is and whether it is safe and how it can be controlled

Reply:
This was a simple proxy used by hardhat-deploy. We have update this now to use `ERC1967Proxy.sol` which is UUPS compatible and is held under the [deployments folder](https://github.com/polymorpher/sms-wallet/tree/miniV0/miniwallet/contracts/deployments)